vminfo_parser/vminfo_parser.py

```diff
@@ -407,50 +427,27 @@ def visualize_os_distribution(
         plt.show(block=True)
         plt.close()
 
-    def visualize_os_distribution_log_scale(
-        self: t.Self,
-        counts: pd.Series,
-        os_names: list[str],
-        environment_filter: t.Optional[str] = None,
-        min_count: int = 500,
-    ) -> None:
-        random_colors = cm.rainbow(np.linspace(0, 1, len(counts)))
-        colors = [const.SUPPORTED_OS_COLORS.get(os, random_colors[i]) for i, os in enumerate(os_names)]
-
-        # ax = counts.plot(kind="barh", rot=45, color=colors)
-        counts.plot(kind="barh", rot=45, color=colors)
-
-        plt.title(f"OS Counts by Environment Type (>= {min_count})")
-        plt.xlabel("Count")
-        plt.ylabel("Operating Systems")
-
-        plt.xscale("log")
-        plt.gca().xaxis.set_major_formatter(ticker.ScalarFormatter())
-
-        if self.analyzer.config.generate_graphs:
-            plt.show(block=True)
-            plt.close()
-
     def visualize_unsupported_os_distribution(self: t.Self, unsupported_counts: pd.Series) -> None:
-        if self.analyzer.config.generate_graphs:
-            random_colors = cm.rainbow(np.linspace(0, 1, len(unsupported_counts)))
-            plt.pie(
-                unsupported_counts,
-                labels=unsupported_counts.index,
-                colors=random_colors,
-                autopct="%1.1f%%",
-            )
-            plt.title("Unsupported Operating System Distribution")
 
-            plt.show(block=True)
-            plt.close()
+        random_colors = cm.rainbow(np.linspace(0, 1, len(unsupported_counts)))
+        plt.pie(
+            unsupported_counts,
+            labels=unsupported_counts.index,
+            colors=random_colors,
+            autopct="%1.1f%%",
+        )
+        plt.title("Unsupported Operating System Distribution")
+
+        plt.show(block=True)
+        plt.close()
 
     def visualize_supported_os_distribution(
         self: t.Self,
         counts: pd.Series,
+        config: Config,
         environment_filter: t.Optional[str] = None,
     ) -> None:
-        if self.analyzer.config.generate_graphs:
+        if config.generate_graphs:
```

Comment:
as we do above for disk space distribution,  we should check this value before calling the function,instead of passing the config in just to check this value.